library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1815,80 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    // #[kani::proof_for_contract(NonNull::len)]
```

Comment:
why is this commented out?

Reply:
cause you mentioned below, so i kept the original version using comments: 

Length is a safe operation, so you should be able to generate a non-deterministic NonNull. I would suggest you to do the following:

Create an implementation of kani::Arbitrary for NonNull<T>.